girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -98,3 +101,84 @@ def list_dandisets(self, limit, offset, sort):
         return Folder().find(
             {"parentId": drafts["_id"]}, limit=limit, offset=offset, sort=sort
         )
+
+    @access.public
+    @describeRoute(Description("Global Dandiset Statistics"))
+    def stats(self, params):
+        drafts = get_or_create_drafts_collection()
+        draft_count = Collection().countFolders(drafts)
+
+        # TODO no way to publish
+        published_count = 0
+
+        user_count = User().collection.count_documents({})
+
+        # These aggregate queries are not indexable.
+        # Here are some benchmarks I generated for this endpoint as is:
+        #    0 dandisets  |  0.008207440376281738 seconds
+        # 1000 dandisets  |  0.019745850563049318 seconds
+        # 2000 dandisets  |  0.030080437660217285 seconds
+        # 1000 more dandisets takes about .011 more seconds
+        # Here is the same benchmark but with these aggregate queries removed:
+        #    0 dandisets  |  0.004081225395202637
+        # 1000 dandisets  |  0.008220505714416505
+        # 2000 dandisets  |  0.011307811737060547
+        # 1000 more dandisets takes about 0.0035 more seconds
+
+        species_count = list(
+            Folder().collection.aggregate(
+                [
+                    {"$unwind": "$meta.dandiset.organism"},
```

Comment:
Does adding a `$project` stage at the start make this significantly faster if there are lots of other fields on the dandiset metadata?

Reply:
I first wrote the queries, then did the benchmarking before doing any optimizations. Unless there are a vastly unexpected number of dandisets added in the near future, optimization now seemed premature.
`$project` doesn't work with indices according to the docs, but it's possible that it would speed up the pipeline to have less data. I can experiment if you think it's important at this stage.